src/commands/runs/ls.ts

```diff
@@ -0,0 +1,283 @@
+	const userInfo = await getLocalUserInfo();
+	const usernameOrId = userInfo.username || (userInfo.id as string);
+	const localConfig = getLocalConfig(process.cwd()) || {};
+
+	// Full ID
+	if (providedActorNameOrId?.includes('/')) {
+		const actor = await client.actor(providedActorNameOrId).get();
+		if (!actor) {
+			return {
+				valid: false as const,
+				reason: `Actor with ID "${providedActorNameOrId}" was not found`,
+			};
+		}
+
+		return {
+			valid: true as const,
+			userFriendlyId: `${actor.username}/${actor.name}`,
+			id: actor.id,
+		};
+	}
+
+	// Try fetching Actor directly by name/id
+	if (providedActorNameOrId) {
+		const actorById = await client.actor(providedActorNameOrId).get();
+
+		if (actorById) {
+			return {
+				valid: true as const,
+				userFriendlyId: `${actorById.username}/${actorById.name}`,
+				id: actorById.id,
+			};
+		}
+
+		const actorByName = await client.actor(`${usernameOrId}/${providedActorNameOrId.toLowerCase()}`).get();
+
+		if (actorByName) {
+			return {
+				valid: true as const,
+				userFriendlyId: `${actorByName.username}/${actorByName.name}`,
+				id: actorByName.id,
+			};
+		}
+
+		return {
+			valid: false as const,
+			reason: `Actor with name or ID "${providedActorNameOrId}" was not found`,
+		};
+	}
+
+	if (localConfig.name) {
+		const actor = await client.actor(`${usernameOrId}/${localConfig.name}`).get();
+
+		if (!actor) {
+			return {
+				valid: false as const,
+				reason: `Actor with name "${localConfig.name}" was not found`,
+			};
+		}
+
+		return {
+			valid: true as const,
+			userFriendlyId: `${actor.username}/${actor.name}`,
+			id: actor.id,
+		};
+	}
+
+	return {
+		valid: false as const,
+		reason: 'Unable to detect what Actor to create a build for',
+	};
+}
+
+function prettyPrintStatus(status: string) {
+	switch (status) {
+		case 'READY':
+			return chalk.green('Ready');
+		case 'RUNNING':
+			return chalk.blue('Running');
+		case 'SUCCEEDED':
+			return chalk.green('Succeeded');
+		case 'FAILED':
+			return chalk.red('Failed');
+		case 'ABORTING':
+			return chalk.yellow('Aborting');
+		case 'ABORTED':
+			return chalk.red('Aborted');
+		case 'TIMING-OUT':
+			return chalk.yellow('Timing Out');
+		case 'TIMED-OUT':
+			return chalk.red('Timed Out');
+		default:
+			return chalk.gray(
+				(status as string)
+					.split('-')
+					.map((part) => part[0].toUpperCase() + part.slice(1).toLowerCase())
+					.join(' '),
+			);
+	}
+}
+
+const ShortDurationFormatter = new SapphireDurationFormatter({
+	[TimeTypes.Day]: {
+		DEFAULT: 'd',
+	},
+	[TimeTypes.Hour]: {
+		DEFAULT: 'h',
+	},
+	[TimeTypes.Minute]: {
+		DEFAULT: 'm',
+	},
+	[TimeTypes.Month]: {
+		DEFAULT: 'M',
+	},
+	[TimeTypes.Second]: {
+		DEFAULT: 's',
+	},
+	[TimeTypes.Week]: {
+		DEFAULT: 'w',
+	},
+	[TimeTypes.Year]: {
+		DEFAULT: 'y',
+	},
+});
+
+// END of TODO
+
+const multilineTimestampFormatter = new Timestamp(`YYYY-MM-DD[\n]HH:mm:ss`);
+const tableFactory = (compact = false) => {
```

Comment:
maybe we could find some abstraction for this, e.g. so we don't need to repeat `cyan` N times, i can imagine this will used in quite a lot of commands

Reply:
I plan on working on this too, because we also want the tables to look good at 80 width -w-, I will probably build a quick-n-dirty abstraction over cli-table so we can define what columns we want 100% of the time, what columns we are willing to drop, and handle the shrinking automatically on that side instead of in code like it is rn

Reply:
Did in the last commit! 👀